### ISSUE TITLE
Deleting execution from api is not always working (random errors)

### ISSUE THREAD

**Issue by katanafleet** [rundeck/rundeck#1380]
Hello,

I'am using Rundeck api to automate purge of old job executions, project by project, on Rundeck 2.4.2-1.
I'm getting some sequences of my API calls which returns a Server Error. 

[ERROR] Error on method DELETE for https://myrundeck/rundeck/api/12/execution/8353 : 500 Internal Server Error

Found this error on stdout of my tomcat container running  rundeck 
```
ERROR  error, Failed to delete execution: null
ERROR  error, Failed to delete execution 8353
groovy.lang.MissingPropertyException: No such property: project for class: rundeck.services.ExecutionService
        at rundeck.services.ExecutionService$_deleteExecution_closure27.doCall(ExecutionService.groovy:1193)
        at rundeck.services.ExecutionService.deleteExecution(ExecutionService.groovy:1191)
        at rundeck.controllers.ExecutionController.apiExecutionDelete(ExecutionController.groovy:1266)
        at com.codahale.metrics.servlet.AbstractInstrumentedFilter.doFilter(AbstractInstrumentedFilter.java:97)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:722)

```

Trying again with a rest client the execution is however well deleted  ... (i'am getting a 404)

**Comment by rmbolger** [comment]
I'm doing the same thing and also on 2.4.2-1 and also getting sporadic failures deleting executions. Mine seems to be tied to specific executions.  My script is trying to delete groups of 10 at a time using the bulk delete (/api/12/executions/delete).  And the same set of IDs fails with no error message in the json output.  Deleting them also fails from the web UI with the following error in server.log:

    ERROR ExecutionService: Failed to delete execution 938
    groovy.lang.MissingPropertyException: No such property: adhocRemoteString for class: rundeck.PluginStep
            at rundeck.services.ExecutionService.createExecutionLogMDCdata(ExecutionService.groovy:610)
            at rundeck.services.ExecutionService.logExecutionLog4j(ExecutionService.groovy:571)
            at rundeck.services.ExecutionService.deleteExecution(ExecutionService.groovy:1182)
            at rundeck.controllers.ExecutionController$_closure4_closure30.doCall(ExecutionController.groovy:154)
            at rundeck.controllers.ExecutionController$_closure4.doCall(ExecutionController.groovy:139)
            at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:565)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1360)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:477)
            at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:119)
            at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:539)
            at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:227)
            at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1031)
            at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:406)
            at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:186)
            at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:965)
            at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:117)
            at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:288)
            at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:115)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at com.codahale.metrics.servlet.AbstractInstrumentedFilter.doFilter(AbstractInstrumentedFilter.java:97)
            at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1331)
            at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:477)
            at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:119)
            at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:486)
            at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:227)
            at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1031)
            at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:406)
            at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:186)
            at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:965)
            at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:117)
            at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:111)
            at org.eclipse.jetty.server.Server.handle(Server.java:349)
            at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:449)
            at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.content(AbstractHttpConnection.java:925)
            at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:857)
            at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
            at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:76)
            at org.eclipse.jetty.io.nio.SslConnection.handle(SslConnection.java:191)
            at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:609)
            at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:45)
            at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:599)
            at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:534)
            at java.lang.Thread.run(Thread.java:745)

**Comment by gschueler** [comment]
second error was fixed in edaaaae3 (rundeck 2.5.2)

**Comment by rmbolger** [comment]
Awesome. Many thanks.
